test/services/workflow_executions/cleanup_service_test.rb

```diff
@@ -47,15 +54,95 @@ def setup # rubocop:disable Metrics/MethodLength
         @output_file_blob.download
         @input_file_blob.download
         @samplesheet_file_blob.download
+        @unrelated_file_blob.download
       end
 
       @workflow_execution = WorkflowExecutions::CleanupService.new(@workflow_execution, @user, {}).execute
 
       assert_raises(ActiveStorage::FileNotFoundError) { @output_file_blob.download }
       assert_raises(ActiveStorage::FileNotFoundError) { @input_file_blob.download }
       assert_raises(ActiveStorage::FileNotFoundError) { @samplesheet_file_blob.download }
+      assert_nothing_raised { @unrelated_file_blob.download }
 
       assert @workflow_execution.cleaned?
     end
+
+    test 'do not clean an already cleaned workflow execution' do
+      @workflow_execution.cleaned = true
+
+      assert_nothing_raised do
+        @output_file_blob.download
+        @input_file_blob.download
+        @samplesheet_file_blob.download
+        @unrelated_file_blob.download
+      end
+
+      @workflow_execution = WorkflowExecutions::CleanupService.new(@workflow_execution, @user, {}).execute
+
+      assert_nothing_raised do
+        @output_file_blob.download
+        @input_file_blob.download
+        @samplesheet_file_blob.download
+        @unrelated_file_blob.download
+      end
```

Comment:
For this test and the ones below, we should assert that the workflow execution is marked as `cleaned` after the cleanup service is called

Reply:
Added in https://github.com/phac-nml/irida-next/pull/658/commits/f271c7de42405ee78590a231b86496f64f6ad346